### ISSUE TITLE
Make Bforartists addons "Official" 

### ISSUE THREAD

**Issue by Draise14** [Bforartists/Bforartists#4121]
![image](https://github.com/Bforartists/Bforartists/assets/35357409/e3ec56f6-3066-4b55-8c22-417e2890fc80)

So that it's easier to discover them and opt-in. 

Though with the extensions platform coming soon.. hmm...

**Comment by ReinerBforartists** [comment]
Good catch :)

**Comment by ReinerBforartists** [comment]
I wonder if it is possible to introduce a BFA category here.

**Comment by Blenux** [comment]
> I wonder if it is possible to introduce a BFA category here.

Why yes it can 😉
![image](https://github.com/Bforartists/Bforartists/assets/25260650/251cc82d-3d81-4c95-8164-2bb3e8e777cf)

**Comment by ReinerBforartists** [comment]
Would imo make more sense. But having them in official would also be nice enough already.

**Comment by Blenux** [comment]
We have to also consider when the extensions will replace the addon section entirely, no?

**Comment by ReinerBforartists** [comment]
Entirely? O_O

**Comment by Blenux** [comment]
Yeah when you enabled the extensions experimental feature, it replaces the addon filter enums with it's extensions one.

https://github.com/Bforartists/Bforartists/assets/25260650/4e069b54-f3e9-4d62-b3e2-4261c30f194a

**Comment by ReinerBforartists** [comment]
Huh, looks scary :D
I have no idea what they want to do there :/

**Comment by Blenux** [comment]
This

https://extensions.blender.org/

https://extensions.blender.org/about/

**Comment by ReinerBforartists** [comment]
I see. But our addons are not at this page ^^

**Comment by Blenux** [comment]
That is true :thinking:

**Comment by Draise14** [comment]
Yeah, once extensions are out of experimental, I got a whole bunch of tasks to do for this... but ehm.... till it is in official let's not worry too much and assume it won't come in 4.1

**Tasks:**
1. Add back the quick category tabs again, like the old addon system. Those were good. 
2. Add more iconography for quick addon read and actions (some missing)
3. Expose some common operators to top level instead of all hidden away. Maybe a new row would be necessary... 
4. Make sure our addons list correctly.... Theoretically this extensions system has local addons like the old way, and the new way.

**Comment by Draise14** [comment]
> > I wonder if it is possible to introduce a BFA category here.
> 
> Why yes it can 😉 ![image](https://private-user-images.githubusercontent.com/25260650/311481732-e0442c09-2672-4bcb-a193-0ddf9e93585b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAxODM2NzEsIm5iZiI6MTcxMDE4MzM3MSwicGF0aCI6Ii8yNTI2MDY1MC8zMTE0ODE3MzItZTA0NDJjMDktMjY3Mi00YmNiLWExOTMtMGRkZjllOTM1ODViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzExVDE4NTYxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUzYWE4MjY4M2YxMWRhZjhjZGY4OWRiNTcyMjQ2MjA5YjFjMTc5ZWY1NGRkOWUyNDhhZDUxMjU3Y2YyZTc5ZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.msuQEpXTsbLQ0XPmDk9PbJgZiKx_EVSjBb-DGVrFcCM)

I guess we can get the same filter for what you did here to go into the Official... and good to go.

**Comment by Draise14** [comment]
How did you do this one @Blenux ?

**Comment by Blenux** [comment]
> How did you do this one @Blenux ?

bl_ui scripts folder, edit the init.py file with the following,

![image](https://github.com/Bforartists/Bforartists/assets/25260650/d7feac9d-4b11-4fa5-bd05-f0db3636b0d1)

and each of the Bforartists add-ons need this added in each of their bl_info, 

"support": "BFA",

Note!, this only works for the current add-ons way, new extensions way is different.

**Comment by Draise14** [comment]
Ah. So I could also add "official" to the add-ons and we good to roll?

The extensions... Let's worry about that once it is out of experimental. Till then....it's a non feature..

**Comment by ReinerBforartists** [comment]
That's imo not really necessary once we would have the Bforartists category. We have just a few addons. It would be a solution when we decide against the bforartists category.

**Comment by Blenux** [comment]
Yeah true, just trying out options :smile:

**Comment by Draise14** [comment]
I love your experiments, meanwhile I'm gonna lump the Bforaritsts addons into the Official tab and will commit and close.

**Comment by Draise14** [comment]
![bforartists_f6xFAuy40T](https://github.com/Bforartists/Bforartists/assets/35357409/e7b7461f-59d3-427e-afc2-ce9f6e8dae4c)

Made our addons show in a Bforartists category too, using the category built in system. There are few enough to know roughly what they are for once you filter these.  

Updating documentation now.

**Comment by Draise14** [comment]
Forgot to pull. One sec.

**Comment by Blenux** [comment]
Houston we have a problem, 

There is a issue with renaming all the addons with new names with bfa and having existing user preferences with the old names, the new ones ain't enabled by default.

![image](https://github.com/Bforartists/Bforartists/assets/25260650/f525fdbe-6ae7-4fd9-9186-d97062d4d966)

**Comment by ReinerBforartists** [comment]
I cannot continue due this one. So let's fix it.

**Comment by ReinerBforartists** [comment]
Hm, very curious. The addons are added/ changed at the right place. But are off O_O

![addonsareoff](https://github.com/Bforartists/Bforartists/assets/13919702/d84ca7d9-c273-4d5a-bd9a-24eccb7dd5d5)

**Comment by ReinerBforartists** [comment]
I see where the trouble is. The new file names conflicts with the old file names in existing user preferences. The user preferences still contains the old addon names. And so we have this conflict here.

Not a happy bunny here. We will have quite a few unhappy old users now. We eagerly need to put a warning to the users to remove their user preferences before using BFA 4.1 then.

**Comment by Draise14** [comment]
I vote keep the new, do publicity campaign / tutorialon release. 

Reverting this will be a mess. We aren't that many old users, and most likely experienced.

**Comment by ReinerBforartists** [comment]
Mh, there must be a way to keep the old names but have the new functionality.

**Comment by Draise14** [comment]
About to turn on computer then will test. I think I know how to reproduce and try fix.

**Comment by Draise14** [comment]
![imagen](https://github.com/Bforartists/Bforartists/assets/35357409/ec3ee916-0d01-4114-bfe5-1f1e9e0de21e)

Ok, reverted, and loaded back old userpref from previous build and.. apparently seems ok. Will commit and push to master, can you test?

**Comment by ReinerBforartists** [comment]
Looking good :)

**Comment by Draise14** [comment]
Great, let's consider this good. The "name" or label of the addon remains the same as in the documentation, and the cateogry, and it is backwards compatible. For the sure it is smooth sailing. For us, well we can live with the file name changes.

**Comment by esnosy** [comment]
i love this <3
